simpleflow/canvas.py

```diff
@@ -44,17 +49,23 @@ def __init__(self,
                  *activities,
                  **options):
         self.activities = []
+        self.workflow_tasks = []  # type: List[WorkflowTask]
         self.max_parallel = options.pop('max_parallel', None)
         self.raises_on_failure = options.pop('raises_on_failure', None)
         self.bubbles_exception_on_failure = options.pop('bubbles_exception_on_failure', True)
         self.extend(activities)
 
     def append(self, submittable, *args, **kwargs):
+        from simpleflow import Workflow
```

Comment:
Why this import is here and not with the others?

Reply:
Circular import otherwise 🙄

Reply:
I tend to think that circular imports are often a hint that we should flatten the files hierarchy; that's why I moved the dispatcher submodule as a top level module a few weeks ago. What do you think?

Reply:
Well, `Workflow.map` creates and submits a `Group`...